### PR TITLE
Handle inapp, forgeground, background notification

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview
Handle all notifications actions
- [x] Press on notification item in app
- [x] Press on notification in foreground vs background mode
- [x] Press on notification when app terminated

## Demo

https://www.loom.com/share/1de3ec413db548409a4730567334d831?sid=600d712c-6fb5-41f4-bd91-bc38e61d5d66

### CODE REVIEW

File: lib/core/utils/navigation_utils.dart [hunk 1]
```diff
@@ -1,32 +1,39 @@
...
+import 'package:auto_route/auto_route.dart';
 import 'package:flutter/material.dart';
 
 class NavigationUtils {
 
-  static handleNotificationNavigate(BuildContext context, NotificationEntities.Notification? notification) {
-    switch(notification?.object_type) {
+  static handleNotificationNavigate(BuildContext context, String? type, String? objectType, String? objectId) {
+    switch(objectType) {
       case "StoreOrder":
```
Comment (mthinh) [inline lib/core/utils/navigation_utils.dart:12]: Please make use of enum 
ex: 
enum NotificationType {
StoreOrder,
....
}

File: lib/core/presentation/pages/home/home_page.dart [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+class HomePage extends StatefulWidget {
   const HomePage({super.key});
 
+  @override
+  State<StatefulWidget> createState() => _HomePageViewState();
+}
+
+class _HomePageViewState extends State<HomePage> {
+  
+  Future<void> setupInteractedMessage() async {
```
Comment (mthinh) [inline lib/core/presentation/pages/home/home_page.dart:18]: Please move these setup to _App widget in app.dart file,  these are global service setup, should not be specific in any business screen

File: lib/core/presentation/pages/home/home_page.dart [hunk 3]
```diff
@@ -15,4 +39,13 @@ class HomePage extends StatelessWidget {
...
     );
   }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      FirebaseService.setContext(context);
+    });
+    setupInteractedMessage();
```
Comment (mthinh) [inline lib/core/presentation/pages/home/home_page.dart:49]: same with above for FirebaseService

File: lib/core/utils/navigation_utils.dart (hunk 1 above)
Comment (sieuhuflit) [reply lib/core/utils/navigation_utils.dart:12]: Resolved

File: lib/core/presentation/pages/home/home_page.dart (hunk 2 above)
Comment (sieuhuflit) [reply lib/core/presentation/pages/home/home_page.dart:18]: Got it, thanks 👍

**Pull request merged**